packages/store-ui/src/atoms/List/List.tsx

```diff
@@ -0,0 +1,56 @@
+import type { HTMLAttributes, ElementType } from 'react'
+import React, { forwardRef } from 'react'
+
+export type ListVariant = 'default' | 'description' | 'ordered' | 'unordered'
+
+const variantToType = (variant: ListVariant) => {
+  switch (variant) {
+    case 'description':
+      return 'dl'
+
+    case 'unordered':
+      return 'ul'
+
+    case 'ordered':
+      return 'ol'
+
+    default:
+      return 'div'
+  }
+}
+
+export interface ListProps extends HTMLAttributes<HTMLElement> {
+  /**
+   * Set the HTML element tag of this component.
+   */
+  as?: ElementType
+  /**
+   * ID to find this component in testing tools (e.g.: cypress, testing library, and jest).
+   */
+  testId?: string
+  /**
+   * The current use case variants for Lists.
+   */
+  variant?: ListVariant
+}
+
+export const List = forwardRef<HTMLElement, ListProps>(function List(
+  {
+    testId = 'store-list',
+    variant = 'default',
+    as: Component = variantToType(variant),
+    ...rawProps
+  },
+  ref
+) {
+  return (
+    <Component
+      ref={ref}
+      data-store-list={variant}
```

Comment:
I haven't reviewed a lot of store UI PRs lately so I may be wrong about this, but I think this deviates from our pattern. What you should really do is to set different data attributes when the variant changes, letting the `data-store-list` value unchanged.

Example of the [price atom](https://github.com/vtex/faststore/blob/81a6a9c4591f59b66d64b9a68f7983851aaa9725/packages/store-ui/src/atoms/Price/Price.tsx#L51):

```tsx
const props = {
  [`data-${variant}`]: true,
  ...rawProps,
}

return (
  <Component ref={ref} data-store-price data-testid={testId} {...props}>
    {formattedPrice}
  </Component>
)
```

If this pattern changed, let me know so we can update the price atom